vext.go

```diff
@@ -69,6 +73,40 @@ func Email() *v.MessageValidator {
 	return v.Is(govalidator.IsEmail).Msg("invalid email")
 }
 
+// EmailNonDisposable is a leaf validator factory used to create a validator, which will
+// succeed when the field's value is a non-disposable email address.
+// It uses the `is-email-disposable` package to check if the email domain is disposable.
+func EmailNonDisposable() *v.MessageValidator {
+	if disposableDomainService == nil {
+		disposableDomainService2, err := domain.New()
+		if err != nil {
+			panic(err)
+		}
+
+		disposableDomainService = disposableDomainService2
+	}
+
+	messageValidator := v.MessageValidator{
+		Message: "is disposable e-mail address",
+	}
+
+	messageValidator.Validator = v.Func(func(field *v.Field) v.Errors {
+		value, ok := field.Value.(string)
+		if !ok {
+			return v.NewUnsupportedErrors("EmailNonDisposable", field, "")
+		}
+
+		checkResult := disposableDomainService.Check(value)
+		if checkResult.IsDisposable {
+			return v.NewInvalidErrors(field, messageValidator.Message)
+		}
+
+		return nil
+	})
+
+	return &messageValidator
```

Comment:
How about using `v.Is()` here like this:

```go
isValid := func(value string) bool {
        checkResult := disposableDomainService.Check(value)
        return !checkResult.IsDisposable
}
return v.Is(isValid).Msg("is disposable e-mail address")
```

Perhaps this is more concise? What's your opinion?

Reply:
You are right, it is a lot better actually! I updated accordingly.